test-loop-tests/src/tests/catching_up.rs

```diff
@@ -0,0 +1,515 @@
+        runner.all_validators_accounts().iter().chain(test_accounts.iter()).cloned().collect_vec();
+
+    let num_shards = 4;
+    let epoch_length = 8;
+    let genesis = TestLoopBuilder::new_genesis_builder()
+        .epoch_length(epoch_length)
+        .validators_spec(runner.genesis_validators_spec(seats, seats, seats))
+        .add_user_accounts_simple(&all_accounts, stake)
+        .shard_layout(ShardLayout::multi_shard(num_shards, 3))
+        .build();
+
+    let epoch_config_store = TestEpochConfigBuilder::from_genesis(&genesis)
+        .minimum_validators_per_shard(2)
+        .build_store_for_genesis_protocol_version();
+
+    let block_prod_time = Duration::milliseconds(6000);
+    let mut env = TestLoopBuilder::new()
+        .genesis(genesis)
+        .clients(runner.all_validators_accounts())
+        .epoch_config_store(epoch_config_store)
+        .config_modifier(move |config, _| {
+            config.min_block_production_delay = block_prod_time;
+            config.max_block_production_delay = 3 * block_prod_time;
+            config.max_block_wait_delay = 3 * block_prod_time;
+        })
+        .build()
+        .warmup();
+
+    for node_datas in &env.node_datas {
+        let peer_actor_handle = node_datas.peer_manager_sender.actor_handle();
+        let peer_actor = env.test_loop.data.get_mut(&peer_actor_handle);
+        peer_actor.register_override_handler(Box::new(move |request| -> Option<NetworkRequests> {
+            if let NetworkRequests::PartialEncodedChunkMessage { partial_encoded_chunk, .. } =
+                &request
+            {
+                if skip_24 {
+                    if partial_encoded_chunk.header.height_created() == 23
+                        || partial_encoded_chunk.header.height_created() == 24
+                    {
+                        return None;
+                    }
+                }
+            }
+            Some(request)
+        }));
+    }
+
+    let client_actor_handle = &env.node_datas[0].client_sender.actor_handle();
+    runner.run_until(
+        &mut env,
+        |test_loop_data| {
+            let client = &test_loop_data.get(client_actor_handle).client;
+            let head = client.chain.head().unwrap();
+            head.height >= send_tx_height
+        },
+        block_prod_time * send_tx_height as i32,
+    );
+
+    let clients = env
+        .node_datas
+        .iter()
+        .map(|test_data| &env.test_loop.data.get(&test_data.client_sender.actor_handle()).client)
+        .collect_vec();
+
+    let mut tx_count = 0;
+    let anchor_hash = get_anchor_hash(&clients);
+    for (i, sender) in test_accounts.iter().enumerate() {
+        for (j, receiver) in test_accounts.iter().enumerate() {
+            let mut amount = (((i + j + 17) * 701) % 42 + 1) as u128;
+            if non_zero {
+                if i > j {
+                    amount = 2;
+                } else {
+                    amount = 1;
+                }
+            }
```

Comment:
Why is non_zero called that and what does it have to do with amount? It doesn't seem like amount is zero in either case.

Reply:
I renamed it to change_balances - IIUC that's essentially what it's doing, since i -> j and j -> i amounts will be different.